examples/DrawXNative/DrawXShared/RealmDraw.cs

```diff
@@ -0,0 +1,572 @@
+using Realms;
+using Realms.Sync;
+using SkiaSharp;
+
+namespace DrawXShared
+{
+    /***
+     Class that does almost everything for the demo that can be shared.
+     It combines drawing with logging in and connecting to the server.
+
+     DrawXSettingsManager provides a singleton to wrap settings and their local Realm.
+
+    Login
+    -----
+    Credentials come from DrawXSettings.
+    It is the responsibility of external GUI classes to get credentials entered and delay
+    starting RealmDraw until connection is made.
+
+    Drawing
+    -------
+    There are three components to drawing:
+    - the background images for "controls"
+    - previously drawn completed, paths which will no longer grow,
+    - the currently growing paths (one per app updating the shared Realm).
+
+    However, with the paths, if we only worry about drawing _added_ or _changed_ paths
+    then we don't care if they are completed or not, just what more we have to draw of them.
+
+    Caching and Responsiveness
+    --------------------------
+    Ideally, we want to have almost all the drawn content cached in a bitmap for redisplay, 
+    and only draw new line segments for each added point. It is relatively easy to optimise for
+    local draw updates because we know when touched that we are drawing a single added segment.
+
+    There are two distinct patterns which can cause a poor demo with synchronised Draw.
+    - A long, winding path being updated - there may be enough lag that we have to add more than 
+      one point to it locally but we don't want to redraw the whole thing from scratch.      
+    - Many small, single "dab" strokes being drawn, say from someone tapping a display, 
+      which mean we have, at least, a TouchesBegan and TouchesEnded and probably AddPoint in between.
+    
+    We make use of a non-persistent field to help a given Draw differentiate which 
+    points it has processed - in the case of an incoming path it may have multiple points 
+    we have not yet seen.
+
+    Most importantly, to get the fastest possible response as the user moves their finger,
+    we draw the local line immediately as a continuation of the path they started drawing 
+    earlier. (See _currentlyDrawing)
+    
+    */
+    public class RealmDraw
+    {
+        private const float NORMALISE_TO = 4000.0f;
+        private const float PENCIL_MARGIN = 4.0f;
+        private const float INVALID_LAST_COORD = -1.0f;
+        private float _lastX = INVALID_LAST_COORD;
+        private float _lastY = INVALID_LAST_COORD;
+
+        #region Synchronised data
+        private Realm _realm;
+        private RealmResults<DrawPath> _allPaths;  // we observe all and filter based on changes
+        #endregion
+
+        #region GUI Callbacks
+        internal Action RefreshOnRealmUpdate { get; set; }
+
+        internal Action CredentialsEditor { get; set; }
+        #endregion
+
+        #region DrawingState
+        private bool _isDrawing = false;
```

Comment:
Actually there is according to Xamarin Studio - I added init when it refused to allow other code. Maybe this one is an exception.